app/models/user.rb

```diff
@@ -19,8 +19,8 @@ def set_token
     token = generate_token
     # TODO: set environment URLs
     response = RestClient.post 'http://localhost:3001/users', user: { token: token }
-    if response == 201
-      self.token = token
+    if response.code == 201
+      self.update_attribute(:token, token)
```

Comment:
The `.code` is mandatory to access the response's code (201 hopefully), and `.update_attribute` actual edits and *saves* the token attribute.